infrastructure/modules/key_vault/main.tf

```diff
@@ -0,0 +1,88 @@
+locals {
+
+}
+
+data "azurerm_client_config" "current" {}
+
+# https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#security
+data "azurerm_role_definition" "key_vault_administrator" {
+  role_definition_id = "00482a5a-887f-4fb3-b363-3b7fe8e74483"
+}
+
+data "azurerm_resource_group" "key_vault" {
+  name = var.resource_group_name
+}
+
+resource "random_string" "key_vault" {
+  length  = 4
+  lower   = true
+  numeric = false
+  upper   = false
+}
+
+# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/key_vault
+resource "azurerm_key_vault" "key_vault" {
+  name                          = "kvencryption${random_string.key_vault.result}"
```

Comment:
Dette er da et veldig rart keyvault name?

Reply:
ikke uenig, men avventer litt med et navnestandard på globally unqiue names

Reply:
KV navn er globally unique? Kan du legge paa en kommentar som sier det foer linje 25 - er kjekt aa ha naer koden. Og hvis det skal vaere globalt unikt saa kan man nesten like gjerne bare bruke et fullt random navn.

Reply:
greit å ha en noe gjenkjenbart i navnet e.g {known_prefix}{rand}. Ellers legge til en tag som beskriver actual name

Reply:
Tag er god plan uansett. Og tenker i utgangspunktet at RGen er prefixen. Dog det fungerer bare hvis man har faa ting i RGer. En av tingene jeg synes er rotete hos altinn er at man har veldig store RGer, som gjoer at allt i en RG maa ha kontext om hva det er. Tidligere steder jeg har hvert har hatt smaa RGer med 2-6 resurser, som gjoer at naar du star i "resurs register" rg, og ser paa en kv, saa spiller det liten rolle hva den heter.

Reply:
Vi skal få lagt til tags for alle ressurser, men det blir ikke i denne PRen. Sant det, vi vil nok ha en lignende strategi når vi faktisk deployer tjenester. Skal vi bare lage et random navn for key vault?

Reply:
`kv-<random>` - bare for aa holde oss til type-preifix ville jeg foreslaat. Og random-lengden burde vaere lengden paa storage account names (8 hvis jeg husker rett).

Reply:
* storageAccounts	: 3-24	Lowercase letters and numbers.
* vaults	global	3-24	Alphanumerics and hyphens. Start with letter. End with letter or digit. Can't contain consecutive hyphens.

keyvault = kv{random<6 chars>} ?
storage account =st{random<6 chars>} ?

Reply:
Jau - dog `sa` i stedet for `st` tror jeg

Reply:
Og kanskje `kv-` siden keyvaults kan ha hyphens.

Reply:
> Jau - dog sa i stedet for st tror jeg
standard forkortning for storage account er vel st
https://learn.microsoft.com/en-us/azure/cloud-adoption-framework/ready/azure-best-practices/resource-abbreviations

Reply:
Ah - har vaert borti sa foer, men ikke st. Men hvis st er standard saa helt enig.